go/host/host.go

```diff
@@ -667,9 +668,36 @@ func (a *Node) monitorBlocks() {
 		select {
 		case err := <-subs.Err():
 			log.Error("Restarting L1 block monitoring. Errored with: %s", err)
-			// todo this is a very simple way of reconnecting the node, it might need catching up logic
+			// it's fine to immediately restart the listener, any incoming blocks will be on hold in the queue
 			listener, subs = a.ethClient.BlockListener()
 
+			// catch up any missed blocks
+			var lastBlkNumber *big.Int
+			// get the latest known blk
+			lastBlk, err := a.ethClient.BlockByNumber(lastBlkNumber)
+			if err != nil {
+				log.Panic("%s", err)
```

Comment:
Bit of context on this might be good since we're panicking. "unable to catch up on missed blocks, last known block not found" or something, I know we'll get the line number but might help to point investigator in the right direction.

Reply:
Good point 👍